results.py

```diff
@@ -48,41 +51,87 @@ def __init__(self,
                benchmark: Benchmark,
                trial: int,
                work_dirs: WorkDirs,
-               status: bool = False,
-               error: str = '',
-               full_log: str = '',
-               insight: str = '',
+               compiles: bool = False,
+               compile_error: str = '',
+               compile_log: str = '',
                fuzz_target_source: str = '',
                build_script_source: str = '',
                author: Any = None,
                agent_dialogs: Optional[dict] = None) -> None:
     super().__init__(benchmark, trial, work_dirs, fuzz_target_source,
                      build_script_source, author, agent_dialogs)
-    self.status: bool = status  # Build success/failure.
-    self.error: str = error  # Build error message.
-    self.full_log: str = full_log  # Build full output.
-    self.insight: str = insight  # Reason and fixes for build failure.
+    self.compiles: bool = compiles  # Build success/failure.
```

Comment:
do we need the type annotations on these lines?

Reply:
I'm not sure if I understand this correctly, but I think these annotations will be helpful to the linters when the variables are used elsewhere?
For local variables, they are not necessary.

Reply:
Sorry I meant: would it be better to define these at a class level?

i.e.

```python

class BuildResult(Result):
  compiles: bool 
  compile_error: str
```

etc. IMO it's a bit easier to find this way.

Reply:
Like this?
https://github.com/google/oss-fuzz-gen/pull/623/commits/c00fbdbaf48008cbb68110bd17aa0eb9547ab3aa